### PR TITLE
Added Resize and Rebuild tests

### PR DESCRIPTION
**Pull request by dwalleck**

### CODE REVIEW

File: etc/stacktester.cfg [hunk 1]
```diff
@@ -3,8 +3,9 @@ host=127.0.0.1
 port=8774
 user=admin
 base_url=v1.1/
-api_key=d9DBcehAEiPAEoAZAb1e
+api_key=9wCAZ0mQdOLu7E7dFTay
```
Comment (throughnothing) [inline etc/stacktester.cfg]: can you change this to just 'admin_key' as a default, and then we can stop comitting it in the future.  You can stage just certain lines in a file so that you can commit the line below without having to commit this stuff each time.  We'll just need to watch to stop comitting this.

File: stacktester/tests/test_server_actions.py [hunk 2]
```diff
@@ -20,9 +20,12 @@ import unittest2 as unittest
 
 
 class ServerActionsTest(unittest.TestCase):
-
+    
+    multi_node = openstack.Manager().config.env.multi_node
```
Comment (throughnothing) [inline stacktester/tests/test_server_actions.py:25]: do we need this as well as the same line in setUpClass ?

**Comment by throughnothing** [comment]
This prop looks good to me other than those few issues I pointed out above.

File: etc/stacktester.cfg (hunk 1 above)
Comment (blamarvt) [inline etc/stacktester.cfg]: Yeah, just need to not commit changes to this file anymore.

Comment (bcwaldon) [inline etc/stacktester.cfg]: I think this is covered in Will's pull request.

File: stacktester/tests/test_server_actions.py [hunk 3]
```diff
@@ -56,9 +56,7 @@ class ServerActionsTest(unittest.TestCase):
         })
         response, body = self.os.nova.request(
             'DELETE',
-            '/servers/%s' % self.server_id,
-            body=body)
-        self.assertEqual('204', response['status'])
```
Comment (bcwaldon) [inline stacktester/tests/test_server_actions.py:61]: Why was this removed?

File: stacktester/tests/test_server_actions.py [hunk 4]
```diff
@@ -37,9 +40,6 @@ class ServerActionsTest(unittest.TestCase):
 
         response, body = self.os.nova.request(
             'POST', '/servers', body=post_body)
-        # KNOWN-ISSUE lp802621
-        #self.assertEqual('202', response['status'])
```
Comment (bcwaldon) [inline stacktester/tests/test_server_actions.py:41]: Why was this removed?

File: stacktester/config.py [hunk 5]
```diff
@@ -60,6 +60,11 @@ class EnvironmentConfig(object):
         """ valid flavorRef to use """
         return self.get("flavor_ref", 1);
 
+    @property
+    def multi_node(self):
+        """ Does the test environment have more than one compute node """
+        return self.get("multi_node", 'false');
```
Comment (bcwaldon) [inline stacktester/config.py]: Can you make this return a boolean? I think it would be easier to use other places in the code

File: stacktester/tests/test_server_actions.py [hunk 6]
```diff
@@ -113,3 +113,80 @@ class ServerActionsTest(unittest.TestCase):
...
+                'name' : 'testserver',
+                'imageRef' : 2,
+                'flavorRef' : self.flavor_ref,
+            }
+        })
+        
+        response, body = self.os.nova.request(
+            'POST', '/servers', body=post_body)
+        
+        # KNOWN-ISSUE lp?
```
Comment (bcwaldon) [inline stacktester/tests/test_server_actions.py:138]: Can you document the correct issue here?

File: stacktester/tests/test_server_actions.py [hunk 7]
```diff
@@ -113,3 +113,80 @@ class ServerActionsTest(unittest.TestCase):
...
+        post_body = json.dumps({
+            'server' : {
+                'name' : 'testserver',
+                'imageRef' : 2,
+                'flavorRef' : self.flavor_ref,
+            }
+        })
+        
+        response, body = self.os.nova.request(
+            'POST', '/servers', body=post_body)
```
Comment (bcwaldon) [inline stacktester/tests/test_server_actions.py]: Let's not break these arguments across two lines here.

File: stacktester/tests/test_server_actions.py [hunk 8]
```diff
@@ -113,3 +113,80 @@ class ServerActionsTest(unittest.TestCase):
...
+        
+        # KNOWN-ISSUE lp?  
+        # self.assertEqual('202', response['status'])
+        self.os.nova.wait_for_server_status(self.server_id, 'REBUILD')
```
Comment (bcwaldon) [inline stacktester/tests/test_server_actions.py]: Can you wrap these two lines with a try/except?

try:
    self.os.nova.wait_for_server_status(self.server_id, 'REBUILD')
except exceptions.TimeoutException:
    self.fail("Server failed to change status to REBUILD")

Same goes for the ACTIVE status below

File: stacktester/tests/test_server_actions.py [hunk 9]
```diff
@@ -113,3 +113,80 @@ class ServerActionsTest(unittest.TestCase):
...
+            'revertResize' : 'null'
+        })
+
+        response, body = self.os.nova.request(
+            'POST', '/servers/%s/action' % self.server_id, body=post_body)
+        self.assertEqual('202', response['status'])
+        self.os.nova.wait_for_server_status(self.server_id, 'ACTIVE')
+        resp, body = self.os.nova.request('GET', '/servers/%s' % self.server_id)
+        data = json.loads(body)        
+        self.assertEqual(self.flavor_refra, data['server']['flavorRef'])
```
Comment (bcwaldon) [inline stacktester/tests/test_server_actions.py]: Should this be self.flavor_ref, not refra?

File: etc/stacktester.cfg (hunk 1 above)
Comment (dwalleck) [inline etc/stacktester.cfg]: This line was changed as a byproduct of my testing, but I needed to commit this file to add a new configuration

Comment (bcwaldon) [inline etc/stacktester.cfg]: Daryl: I didn't even think about that, thanks for pointing it out. I think this is a non-issue for now. We're good :)

File: stacktester/tests/test_server_actions.py (hunk 2 above)
Comment (dwalleck) [inline stacktester/tests/test_server_actions.py:25]: No, it can't be from what I've seen. The skip decorator has no reference to self, so this was included outside of setup.

File: etc/stacktester.cfg (hunk 1 above)
Comment (throughnothing) [inline etc/stacktester.cfg]: You can stage certain lines in a file and not stage other lines with git.  You could add the changes you needed without that line, but nonetheless, this is fine :)

File: stacktester/tests/test_server_actions.py (hunk 3 above)
Comment (dwalleck) [inline stacktester/tests/test_server_actions.py:61]: I removed the assertion for the response code of the delete. This response code is already checked in delete server test, so including it again is a duplication of work. This would make the test code needlessly longer

File: stacktester/tests/test_server_actions.py (hunk 2 above)
Comment (throughnothing) [inline stacktester/tests/test_server_actions.py:25]: hmmm, then do we need it in setUp also?  I'd like to find a way to not have this same thing in multiple places.

File: stacktester/tests/test_server_actions.py (hunk 4 above)
Comment (dwalleck) [inline stacktester/tests/test_server_actions.py:41]: I removed the assertion for the response code of the delete. This response code is already checked in delete server test, so including it again is a duplication of work. This would make the test code needlessly longer

Comment (bcwaldon) [inline stacktester/tests/test_server_actions.py:41]: Cool. Just wanted to hear the justification.

File: stacktester/config.py (hunk 5 above)
Comment (dwalleck) [inline stacktester/config.py]: I tried, but the config was not being pulled in already. I was asked to get my branch in asap, so I kept this as a stopgap. This will be addressed in the future

File: stacktester/tests/test_server_actions.py (hunk 6 above)
Comment (dwalleck) [inline stacktester/tests/test_server_actions.py:138]: I asked this morning to see if this was a bug, but I haven't had any replies yet to my question. Again, to allow my code to get merged in a timely manner, I included this as a placeholder

File: stacktester/tests/test_server_actions.py (hunk 8 above)
Comment (dwalleck) [inline stacktester/tests/test_server_actions.py]: I can, but could you tell me why? To specifically catch the timeout? I'm not sure why it wouldn't hurt to allow the exception to flow up

File: stacktester/tests/test_server_actions.py (hunk 9 above)
Comment (dwalleck) [inline stacktester/tests/test_server_actions.py]: Typo, thanks

File: stacktester/config.py (hunk 5 above)
Comment (bcwaldon) [inline stacktester/config.py]: All you have to do is add  != 'false' to the end of the return line (and you don't need semicolons ;) )

File: stacktester/tests/test_server_actions.py (hunk 8 above)
Comment (bcwaldon) [inline stacktester/tests/test_server_actions.py]: Personally, I don't want to see a traceback for behavior like this in a test. I want to see a message like "Failed to build server". It just makes presentation a bit more meaningful to whomever is running the test.

File: etc/stacktester.cfg (hunk 1 above)
Comment (throughnothing) [inline etc/stacktester.cfg]: This is just for future reference (for me too), I looked up how to do this via the command line (i usually use a gui tool).  If you have multiple changes in a single file, and you only want to commit some of the changes, you can do git add -p etc/stacktester.cfg, and it will ask you y/n for each hunk in the file.  This way, you can say 'n' on the api_key, and y to the other stuff you want.  Just something that may be useful in the future.

File: stacktester/tests/test_server_actions.py (hunk 8 above)
Comment (dwalleck) [inline stacktester/tests/test_server_actions.py]: Fair enough. Given how often we'll be doing actions and waiting for state changes, it'd be nice to factor this block out. I'll make this change and think about other options later

File: stacktester/config.py (hunk 5 above)
Comment (dwalleck) [inline stacktester/config.py]: Sorry, I had mis-read your comment. I see what you're saying now

File: stacktester/tests/test_server_actions.py (hunk 8 above)
Comment (blamarvt) [inline stacktester/tests/test_server_actions.py]: wait_for_server_status could just raise AssertionError, which is what self.fail does.
